runner/progress.py

```diff
@@ -20,14 +32,37 @@ def __init__(self, num_tests: int):
         self.update_idletasks()
 
     def notify_running(self, example: TestExample):
-        self._label.config(text=f"Running {example.dataset_name} ({example.example_id})")
-        self.update_idletasks()
+        self._at = max(self._at, example.start_token)
+        self._memory_span = self._memory_span or example.dataset_generator.memory_span
+        self._test_info[example.unique_id] = dict(start=example.start_token, span=self._memory_span)
```

Comment:
If `_memory_span` is an attribute of the class, its going to be same for all elements of `_test_info`

Reply:
Yes, but it's updated when the test actually finishes, which might be earlier than that. It's probably not worth the extra thought, but it helped me figure it out.

Reply:
(Maybe I wasn't clear enough) `info["span"]` is updated when the test finishes, so that it actually shows a 100% complete when the progress bar is recomputed. Otherwise it would get stuck at 90% or so.

Reply:
You mean at line 46? I'm not so sure... Because its updated to be the max value of the current token, and when it starts, vs the span. Since all tests come in under the memory span, `info["span"]` is just the memory span value again.

Reply:
Ah right! I should have taken the minimum of both. Noted! 👌🏾

Reply:
In the end I just took the actual span. After all, it will be the more accurate regardless it being shorter or longer.